lexer/lexer.go

```diff
@@ -72,12 +72,18 @@ func (l *Lexer) NextToken() *token.Token {
 		} else {
 			return token.EOFT
 		}
+	case '.':
+		if nextChar == '.' { // DOTDOT
+			l.pos++
+			return token.ConstantTokenChar2(ch, nextChar)
+		}
+		// number can start with . eg .5
+		return token.Intern(l.readNumber(ch))
 	default:
 		switch {
```

Comment:
I would have expected something like that


```suggestion
	case '.':
		if nextChar == '.' { // DOTDOT
			l.pos++
			return token.ConstantTokenChar2(ch, nextChar)
		}
		fallthrough
 	default:
 		switch {
```

or


```suggestion

	default:
		if ch == '.' && nextChar == '.' { // DOTDOT
			l.pos++
			return token.ConstantTokenChar2(ch, nextChar)
		}
 	
 		switch {
```

Reply:
Please add tests according to this

Reply:
there are tests already for .3 etc

Reply:
first proposal is worse as it does a check for letter when we already know there is a `.`

and same for 2nd one, also all the other 2 token check are in the main switch, not in default/fallback
the start with digit or number is I think actually clearer than with the before `|| ch = '.'`

and I don't want to have `ch == '.'` checked twice, I prefer to repeat token.Intern(l.readNumber(ch))

Reply:
I might be wrong in the way I reported it

But my point was that code let me think there could be a problem with

1.a
.a
1.-
4. (4 dot space)

The current code seem to me to be:
- we found a dot
- let's read the number that follows

But what comes next might not be a number

Maybe it's something that readNumber handles well. But with previous code the one in the default case, the type was checked

Reply:
> we found a dot let's read the number that follows

that's what the old code did too, if we agree that new code and old code are exactly the same except when `..` is found we can then separate the issue of what happens with invalid numbers from the structure of the code

old (grol 0.31.0) code:
```
$ .a
11:49:35.063 [CRI] parser has 1 error(s)
11:49:35.063 [ERR] parser error: could not parse "." as float
```
same with new code and it'll be the same for all your examples as they don't involve 2 `..` 
(and again there are lots of test about these, like 4. space is 4.0 float and spaces are ignored, etc)